middleware/modelExport.js

```diff
@@ -162,151 +169,130 @@ module.exports = {
-                .catch(cb);
-
-        });
-        asyncFn.push(helpers.getForm);
-        asyncFn.push(helpers.getList);
-
-        // get the actual export object
-        asyncFn.push(function (filters, form, list, callback) {
-
-            // retrieve the export
-            var _export = getExport(list.export);
-            _export.fields = {};
-
-            return callback(null, filters, form, list, _export);
-
-        });
-
-        asyncFn.push(function (filters, form, list, exportObj, cb) {
-            return req.linz.model.getLabels((err, labels) => cb(err, filters, form, list, exportObj, labels));
-        });
-
-        async.waterfall(asyncFn, function (err, filters, form, list, exportObj, labels) {
-
-            if (err) {
-                return next(err);
-            }
-
-            var fields = req.body.selectedFields.split(','),
-                refFieldNames = [],
-                filterFieldNames = [];
-
-            fields.forEach(function (fieldName, index, arr) {
+                    return refFieldNames.push(fieldName);
 
-                // remove any property that doesn't exist in the model
-                if (!Model.schema.tree[fieldName]) {
-                    return arr.splice(index, 1);
-                }
+                });
 
-                // check if there any ref fields selected to be exported
-                if(!Model.schema.tree[fieldName].ref) {
-                    return;
+                // If _id is not included, exclude it from the export;
+                if (!fields.includes('_id')) {
+                    filterFieldNames.push('-_id');
                 }
 
-                return refFieldNames.push(fieldName);
+                filterFieldNames = filterFieldNames.concat(fields);
 
-            });
+                // pipe data to response stream
+                req.linz.model.getQuery(req, filters, function getQuery (err, query) {
 
-            // check if _id is excluded
-            if (exportObj.exclusions.indexOf('_id') >= 0) {
-                filterFieldNames.push('-_id');
-            }
-
-            filterFieldNames = filterFieldNames.concat(fields);
-
-            // pipe data to response stream
-            req.linz.model.getQuery(req, filters, function getQuery (err, query) {
+                    if (err) {
+                        return next(err);
+                    }
 
-                if (err) {
-                    return next(err);
-                }
+                    req.linz.model.listQuery(req, query, (listQueryErr, listQuery) => {
 
-                req.linz.model.listQuery(req, query, (listQueryErr, listQuery) => {
+                        if (listQueryErr) {
+                            return next(listQueryErr);
+                        }
 
-                    if (listQueryErr) {
-                        return next(listQueryErr);
-                    }
+                        const exclusions = exportObj.exclusions.split(',');
```

Comment:
i think this variable can be removed

Reply:
@DinoLeung, good pick up, I've just fixed that up.